src/app/api/services/dot-license/dot-license.service.spec.ts

```diff
@@ -1,80 +1,85 @@
+        this.backend = this.injector.get(ConnectionBackend) as MockBackend;
+        this.backend.connections.subscribe((connection: any) => (this.lastConnection = connection));
     });
 
     it('should return a false response because license is 100 = Community', () => {
-        let result: boolean;
-        dotLicenseService.isEnterpriseLicense().subscribe((res) => (result = res));
+        let result;
+        this.dotLicenseService.isEnterpriseLicense().subscribe((res) => (result = res));
+        this.lastConnection.mockRespond(
+            new Response(
+                new ResponseOptions({
+                    body: {
+                        entity: {
+                            config: {
+                                license: {
+                                    level: 100
+                                }
+                            }
+                        }
+                    }
+                })
+            )
+        );
 
         expect(result).toBe(false);
     });
 
     it('should return a true response because license is equal to 200', () => {
         let result: boolean;
-
-        spyOn(dotCmsConfig, 'getConfig').and.returnValue(
-            Observable.of({
-                license: {
-                    level: 200
-                }
-            })
+        this.dotLicenseService.isEnterpriseLicense().subscribe((res) => (result = res));
+        this.lastConnection.mockRespond(
+            new Response(
+                new ResponseOptions({
+                    body: {
+                        entity: {
+                            config: {
+                                license: {
+                                    level: 200
+                                }
+                            }
+                        }
+                    }
+                })
+            )
         );
 
-        dotLicenseService.isEnterpriseLicense().subscribe((res) => (result = res));
         expect(result).toBe(true);
     });
 
     it('should return a true response because license is equal to 400', () => {
         let result: boolean;
-
-        spyOn(dotCmsConfig, 'getConfig').and.returnValue(
-            Observable.of({
-                license: {
-                    level: 400
-                }
-            })
+        this.dotLicenseService.isEnterpriseLicense().subscribe((res) => (result = res));
+        this.lastConnection.mockRespond(
+            new Response(
+                new ResponseOptions({
+                    body: {
+                        entity: {
+                            config: {
+                                license: {
+                                    level: 400
+                                }
+                            }
+                        }
+                    }
+                })
+            )
         );
 
-        dotLicenseService.isEnterpriseLicense().subscribe((res) => (result = res));
```

Comment:
maybe for we should have a test to check that a request is sent to the BE each time

Reply:
yup